src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -50,9 +50,26 @@ class RsExtractFunctionConfig private constructor(
 ) {
     val signature: String
         get() {
-            var signature = "fn $name(${if (needsSelf) "self" else ""})"
-            if (returnValue != null) {
-                signature += " -> ${returnValue.type}"
+            var signature = "fn $name"
+            val arguments = "(${if (needsSelf) "self" else ""})"
+            signature += if (returnValue != null) {
+                var type = returnValue.type.toString()
+                val unknowns = Regex("<unknown>").findAll(type)
```

Comment:
Ouch, that seems gross... Could we check this in a more structured way, rather then relying on a `toString` representation?

Reply:
I tried to find a fix solve for it. Besides, i also don't like the complete signature function.

Reply:
There's `org.rust.ide.presentation.TyKt#getShortPresentableText`, and I think we can make `Ty.insertionSafeText` using the same machinery, which would replace `<unknowns>` with something reasonable.

Reply:
But in general yeah, I think `ExtractFunction` can use some refactoring and restructuring: this is really complex stuff, with a ton of edge cases, so code must me architectured with care to avoid drowning in the error handing and such.  

We might add some infrastructure for showing error popup with "Sorry, can't extract this function, because it contains this stuff, which we can't handle yet"

Reply:
We can start with making _ and replace this in the future

Reply:
Sounds reasonable! But we still need to extract this into a separate method, to make it *easy* to replace in future.